ale_linters/java/eclipselsp.vim

```diff
@@ -0,0 +1,92 @@
+" Author: Horacio Sanson <https://github.com/hsanson>
+" Description: Support for the Eclipse language server https://github.com/eclipse/eclipse.jdt.ls
+
+call ale#Set('java_eclipselsp_path', 'eclipse.jdt.ls/org.eclipse.jdt.ls.product/target/repository')
+call ale#Set('java_eclipselsp_executable', 'java')
+
+function! ale_linters#java#eclipselsp#Executable(buffer) abort
+    return ale#Var(a:buffer, 'java_eclipselsp_executable')
+endfunction
+
+function! ale_linters#java#eclipselsp#TargetPath(buffer) abort
+    return ale#Var(a:buffer, 'java_eclipselsp_path')
+endfunction
+
+function! ale_linters#java#eclipselsp#JarPath(buffer) abort
+    let l:path = ale_linters#java#eclipselsp#TargetPath(a:buffer)
+
+    return l:path . '/plugins/org.eclipse.equinox.launcher_1.5.200.v20180922-1751.jar'
+endfunction
+
+function! ale_linters#java#eclipselsp#ConfigurationPath(buffer) abort
+    let l:path = ale_linters#java#eclipselsp#TargetPath(a:buffer)
+
+    if has('win32')
+      return l:path . '/config_win'
+    elseif has('macunix')
+      return l:path . '/config_mac'
+    else
+      return l:path . '/config_linux'
+    endif
+endfunction
+
+function! ale_linters#java#eclipselsp#VersionCheck(version_lines) abort
+    let l:version = []
+
+    for l:line in a:version_lines
+        let l:match = matchlist(l:line, '\(\d\+\)\.\(\d\+\)\.\(\d\+\)')
+
+        if !empty(l:match)
+            let l:version = [l:match[1] + 0, l:match[2] + 0, l:match[3] + 0]
+            break
+        endif
+    endfor
+
+    return l:version
+endfunction
+
+function! ale_linters#java#eclipselsp#Command(buffer) abort
+    let l:executable = ale_linters#java#eclipselsp#Executable(a:buffer)
+
+    if empty(l:executable)
+        return ''
+    endif
+
+    let l:version_lines = split(system(ale#Escape(l:executable) . ' -version'), '\n')
```

Comment:
The system call here we can't merge in, as `system()` calls are blocking. I or someone else will have to first implement support for chaining commands for LSP, which will be pretty complicated, but worth it.

Reply:
If you just need the version to determine if the extra arguments are required, can you just make a config value "using java > 1.8"?

Reply:
Adding a configuration option for this can work in the short term but would not be a proper solution. I myself keep jumping between Java 8, 9, and 11 so I would have to update that configuration when switching projects.

On the other hand this branch works and I do my best to keep it updated with origin/master so is updated. Feel free to use this branch as I do myself. The only drawback is that you may experience some lockup of vim when the system() method is called but if your machine is fast enough it should not be noticeable at all.